tests/unit/test-extension-runners/test.extension-runners.firefox-desktop.js

```diff
@@ -0,0 +1,428 @@
+        // Simulate an older Firefox that will throw this error.
+        installTemporaryAddon: sinon.spy(
+          () => Promise.reject(new RemoteTempInstallNotSupported(''))
+        ),
+      },
+    });
+
+    sinon.spy(params, 'firefoxClient');
+
+    const runnerInstance = new FirefoxDesktopExtensionRunner(params);
+    return runnerInstance.run()
+      .then(makeSureItFails())
+      .catch(onlyInstancesOf(WebExtError, (error) => {
+        assert.equal(remoteFirefox.installTemporaryAddon.called, true);
+        assert.match(error.message, /use --pre-install/);
+      }));
+  });
+
+  async function testBinaryArgs(extensionRunnerParams, expectedBinaryArgs) {
+    const {params} = prepareExtensionRunnerParams({
+      params: {
+        ...extensionRunnerParams,
+      },
+    });
+
+    sinon.spy(params, 'firefoxClient');
+
+    const runnerInstance = new FirefoxDesktopExtensionRunner(params);
+    await runnerInstance.run();
+
+    // $FLOW_IGNORE: ignored 'property not found' on sinon spy.
+    assert.ok(params.firefoxApp.run.calledOnce);
+    // $FLOW_IGNORE: ignored 'property not found' on sinon spy.
+    assert.deepEqual(params.firefoxApp.run.firstCall.args[1].binaryArgs,
+                     expectedBinaryArgs);
+  }
+
+  it('passes -jsconsole when --browser-console is specified', async () => {
+    await testBinaryArgs({
+      browserConsole: true,
+    }, [
+      '-jsconsole',
+    ]);
+  });
+
+  it('passes single url parameter to Firefox when specified', async () => {
+    await testBinaryArgs({
+      startUrl: 'url1',
+    }, [
+      '--url', 'url1',
+    ]);
+  });
+
+  it('passes multiple url parameters to Firefox when specified', async () => {
+    await testBinaryArgs({
+      startUrl: ['url1', 'url2'],
+    }, [
+      '--url', 'url1', '--url', 'url2',
+    ]);
+  });
+
+  it('passes a custom Firefox profile when specified', async () => {
+    const {params} = prepareExtensionRunnerParams({
+      params: {
+        profilePath: '/path/to/profile',
+      },
+    });
+
+    const runnerInstance = new FirefoxDesktopExtensionRunner(params);
+    await runnerInstance.run();
+
+    // $FLOW_IGNORE: ignored 'property not found' on sinon spy.
+    assert.equal(params.firefoxApp.createProfile.called, false);
+    // $FLOW_IGNORE: ignored 'property not found' on sinon spy.
+    assert.ok(params.firefoxApp.copyProfile.calledOnce);
+    // $FLOW_IGNORE: ignored 'property not found' on sinon spy.
+    assert.equal(params.firefoxApp.copyProfile.firstCall.args[0],
+                 params.profilePath);
+  });
+
+  it('keeps changes in custom profile when specified', async () => {
+    const {params} = prepareExtensionRunnerParams({
+      params: {
+        profilePath: '/path/to/profile',
+        keepProfileChanges: true,
+      },
+    });
+
+    const runnerInstance = new FirefoxDesktopExtensionRunner(params);
+    await runnerInstance.run();
+
+    // $FLOW_IGNORE: ignored 'property not found' on sinon spy.
```

Comment:
oof. I think moving to [sinon.assert](https://github.com/mozilla/web-ext/issues/972) methods will make Flow happy.